### PR TITLE
EW-626: Update personenkontext.

### PR DESCRIPTION
**Pull request by mkreuzkam-cap**
# Description
Adds a PUT endpoint to update a personenkontext.

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/EW-626

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/modules/person/api/personenkontext.uc.spec.ts [hunk 1]
```diff
@@ -193,4 +194,61 @@ describe('PersonenkontextUc', () => {
...
+                await expect(updatePersonPromise).rejects.toThrow(error);
+            });
+        });
+
+        describe('when person for personenkontext could not be found', () => {
+            it('should throw Error', async () => {
+                // AI next 13 lines
+                const personenkontextDo: PersonenkontextDo<true> = DoFactory.createPersonenkontext(true);
+                const error: EntityNotFoundError = new EntityNotFoundError('Person');
+                personenkontextServiceMock.updatePersonenkontext.mockResolvedValue({
```
Comment (psachmann) [inline src/modules/person/api/personenkontext.uc.spec.ts:240]: Missing blank line.

**Review by psachmann — CHANGES_REQUESTED** [review]

File: src/modules/person/api/person-api.mapper.profile.spec.ts [hunk 2]
```diff
@@ -199,5 +201,21 @@ describe('PersonApiMapperProfile', () => {
...
         });
+
+        it('should map UpdatePersonenkontextBodyParams to UpdatePersonenkontextDto', () => {
+            expect(() =>
+                sut.map(
+                    {} as UpdatePersonenkontextBodyParams,
+                    UpdatePersonenkontextBodyParams,
+                    UpdatePersonenkontextDto,
+                ),
+            ).not.toThrowError(MappingError);
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/api/person-api.mapper.profile.spec.ts]: Done

File: src/modules/person/api/person-api.mapper.profile.spec.ts [hunk 3]
```diff
@@ -199,5 +201,21 @@ describe('PersonApiMapperProfile', () => {
...
+        });
+
+        it('should map UpdatePersonenkontextDto to PersonenkontextDo', () => {
+            expect(() =>
+                sut.map({} as UpdatePersonenkontextDto, UpdatePersonenkontextDto, PersonenkontextDo),
+            ).not.toThrowError(MappingError);
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/api/person-api.mapper.profile.spec.ts]: Done

File: src/modules/person/domain/personenkontext.service.ts [hunk 4]
```diff
@@ -66,4 +68,45 @@ export class PersonenkontextService {
...
         return result;
     }
+
+    public async updatePersonenkontext(
+        personenkontextDo: PersonenkontextDo<true>,
+    ): Promise<Result<PersonenkontextDo<true>, DomainError>> {
+        const storedPersonenkontext: Option<PersonenkontextDo<true>> = await this.personenkontextRepo.findById(
+            personenkontextDo.id,
+        );
+        if (!storedPersonenkontext) {
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/domain/personenkontext.service.ts]: Done

File: src/modules/person/domain/personenkontext.service.ts [hunk 5]
```diff
@@ -66,4 +68,45 @@ export class PersonenkontextService {
...
+                ),
+            };
+        }
+
+        const newRevision: string = (parseInt(storedPersonenkontext.revision) + 1).toString();
+        const newData: {
+            referrer?: string;
+            personenstatus?: string;
+            jahrgangsstufe?: string;
+            revision: string;
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/domain/personenkontext.service.ts]: Done

File: src/modules/person/domain/personenkontext.service.ts [hunk 6]
```diff
@@ -66,4 +68,45 @@ export class PersonenkontextService {
...
+        } = {
+            referrer: personenkontextDo.referrer,
+            personenstatus: personenkontextDo.personenstatus,
+            jahrgangsstufe: personenkontextDo.jahrgangsstufe,
+            revision: newRevision,
+        };
+        const updatedPersonenkontextDo: PersonenkontextDo<true> = Object.assign(storedPersonenkontext, newData);
+
+        const saved: Option<PersonenkontextDo<true>> = await this.personenkontextRepo.save(updatedPersonenkontextDo);
+        if (!saved) {
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/domain/personenkontext.service.ts:102]: Done

File: src/modules/person/api/personenkontext.uc.spec.ts (hunk 1 above)
Comment (mkreuzkam-cap) [reply src/modules/person/api/personenkontext.uc.spec.ts]: Done

**Review by Fshmit — APPROVED** [review]

File: src/modules/person/domain/personenkontext.service.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (psachmann) [reply src/modules/person/domain/personenkontext.service.ts:102]: const updatedPersonenkontextDo: PersonenkontextDo<true> = Object.assign(storedPersonenkontext, newData);
        const saved: Option<PersonenkontextDo<true>> = await this.personenkontextRepo.save(updatedPersonenkontextDo);

        if (!saved) {
            return { ok: false, error: new EntityCouldNotBeUpdated('Personenkontext', updatedPersonenkontextDo.id) };
        }

        return { ok: true, value: saved };

**Review by psachmann — APPROVED** [review]

**Pull request merged**